v-next/hardhat/src/internal/cli/telemetry/sentry/transport.ts

```diff
@@ -0,0 +1,49 @@
+import type { Event, Response } from "@sentry/node";
+
+import { spawnDetachedSubProcess } from "@ignored/hardhat-vnext-utils/subprocess";
+
+// This class is wrapped in a function to avoid having to
+// import @sentry/node just for the BaseTransport base class
+export async function getSubprocessTransport(): Promise<any> {
+  const { Status, Transports } = await import("@sentry/node");
+
+  class SubprocessTransport extends Transports.BaseTransport {
+    public override async sendEvent(event: Event): Promise<Response> {
+      const extra: { configPath?: string } = event.extra ?? {};
+      const { configPath } = extra;
+
+      // Don't send user's full config path for privacy reasons
+      delete event.extra?.configPath;
+      // We don't care about the verbose setting
+      delete event.extra?.verbose;
+
+      const serializedEvent = JSON.stringify(event);
+
+      const fileExt =
```

Comment:
Explaining this would be helpful

Reply:
[Done here](https://github.com/NomicFoundation/hardhat/pull/5570/commits/480589a11adfeba18855a8be0679e4241aa9f8d8) with also another comment explanation